applications/asset_tracker_v2/src/ext_sensors/ext_sensors_bsec.c

```diff
@@ -133,81 +196,233 @@ static int enable_settings(void)
+	return i2c_write_dt(&bme680_i2c_dev, buf, len + 1);
 }
 
-static int8_t bus_read(uint8_t dev_addr, uint8_t reg_addr, uint8_t *reg_data_ptr, uint16_t len)
+static int8_t bus_read(uint8_t reg_addr, uint8_t *reg_data_ptr, uint32_t len, void *intf_ptr)
 {
-	return i2c_write_read(config.i2c_master, dev_addr, &reg_addr, 1, reg_data_ptr, len);
+	return i2c_write_read_dt(&bme680_i2c_dev, &reg_addr, 1, reg_data_ptr, len);
 }
 
-static int64_t get_timestamp_us(void)
+static void delay_us(uint32_t period, void *intf_ptr)
 {
-	return k_uptime_get() * MSEC_PER_SEC;
+	k_sleep(K_USEC(period));
 }
 
-static void delay_ms(uint32_t period)
-{
-	k_sleep(K_MSEC(period));
-}
-
-static void output_ready(int64_t timestamp, float iaq, uint8_t iaq_accuracy, float temperature,
-			 float humidity, float pressure, float raw_temperature, float raw_humidity,
-			 float gas, bsec_library_return_t bsec_status, float static_iaq,
-			 float co2_equivalent, float breath_voc_equivalent)
+static void output_ready(const bsec_output_t *outputs, uint8_t n_outputs)
 {
 	k_spinlock_key_t key = k_spin_lock(&(ctx.sensor_read_lock));
-
-	ctx.temperature_latest = (double)temperature;
-	ctx.humidity_latest = (double)humidity;
-	ctx.pressure_latest = (double)pressure / 1000;
-	ctx.air_quality_latest = (uint16_t)iaq;
-
+	for (size_t i = 0; i < n_outputs; ++i) {
+		switch (outputs[i].sensor_id) {
+		case BSEC_OUTPUT_IAQ:
+			ctx.air_quality_latest = (uint16_t) outputs[i].signal;
+			LOG_INF("IAQ: %d", ctx.air_quality_latest);
+			break;
+		case BSEC_OUTPUT_SENSOR_HEAT_COMPENSATED_TEMPERATURE:
+			ctx.temperature_latest = (double) outputs[i].signal;
+			LOG_INF("Temp: %.2f", ctx.temperature_latest);
+			break;
+		case BSEC_OUTPUT_RAW_PRESSURE:
+			ctx.pressure_latest = (double) outputs[i].signal / 1000.f;
+			LOG_INF("Press: %.2f", ctx.pressure_latest);
+			break;
+		case BSEC_OUTPUT_SENSOR_HEAT_COMPENSATED_HUMIDITY:
+			ctx.humidity_latest = (double) outputs[i].signal;
+			LOG_INF("Hum: %.2f", ctx.humidity_latest);
+			break;
+		default:
+			LOG_WRN("unknown bsec output id: %d", outputs[i].sensor_id);
+			break;
+		}
+	}
 	k_spin_unlock(&(ctx.sensor_read_lock), key);
 }
 
-static uint32_t state_load(uint8_t *state_buffer, uint32_t n_buffer)
+static size_t sensor_data_to_bsec_inputs(bsec_bme_settings_t sensor_settings,
+					 const struct bme68x_data *data,
+					 bsec_input_t *inputs, uint64_t timestamp_ns)
 {
-	if ((ctx.state_buffer_len > 0) && (ctx.state_buffer_len <= n_buffer)) {
-		memcpy(state_buffer, ctx.state_buffer, ctx.state_buffer_len);
-		return ctx.state_buffer_len;
+	size_t i = 0;
+
+	if (BSEC_INPUT_PRESENT(sensor_settings, BSEC_INPUT_TEMPERATURE)) {
+		inputs[i].sensor_id = BSEC_INPUT_HEATSOURCE;
+		inputs[i].signal = temp_offset;
+		inputs[i].time_stamp = timestamp_ns;
+		i++;
+		inputs[i].sensor_id = BSEC_INPUT_TEMPERATURE;
+		inputs[i].signal = data->temperature;
+	#ifndef BME68X_USE_FPU
+		inputs[i].signal /= 100.0f;
```

Comment:
There is no documentation about this in the bme68x driver. These values are taken from the C++ wrapper code in BSEC2. It is also not immediately obvious by looking at the source of the driver since calculations have been obfuscated.

Reply:
I just validated both build variants manually, but this should be easily testable in CI.

Reply:
example output:
```[00:04:10.939,544] <dbg> ext_sensors_bsec: sensor_data_to_bsec_inputs: Temp offset: 2.00
[00:04:10.939,605] <dbg> ext_sensors_bsec: sensor_data_to_bsec_inputs: Temp: 25.44
[00:04:10.939,605] <dbg> ext_sensors_bsec: sensor_data_to_bsec_inputs: Hum: 14.77
[00:04:10.939,636] <dbg> ext_sensors_bsec: sensor_data_to_bsec_inputs: Press: 100699.00
[00:04:10.939,636] <dbg> ext_sensors_bsec: sensor_data_to_bsec_inputs: Gas: 245660.00
[00:04:10.940,063] <dbg> ext_sensors_bsec: output_ready: IAQ: 50
[00:04:10.940,063] <dbg> ext_sensors_bsec: output_ready: Press: 100699.00
[00:04:10.940,093] <dbg> ext_sensors_bsec: output_ready: Temp: 23.38
[00:04:10.940,124] <dbg> ext_sensors_bsec: output_ready: Hum: 16.74
```

Reply:
@DematteisGiacomo @jorgenmk